pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -659,6 +659,10 @@ func (adc *attachDetachController) GetPluginDir(podUID string) string {
 	return ""
 }
 
+func (adc *attachDetachController) SetKubeletError(err error) {
```

Comment:
having to add this method to unrelated controllers indicates the method probably doesn't belong in the interface it was added to

Reply:
this is an unfortunate remnant of the `VolumeHost` structure. If you look at all the other methods in volume host they are exactly the same, in reality volumehost should *just* be a Kubelet concept but the other volume controllers kind of ended up using it too so they all just implement nil/empty methods. It is a known issue and is due for a refactor but that is outside of the scope of this PR.